specification/applicationinsights/resource-manager/Microsoft.Insights/preview/2020-10-26-preview/diagnosticServicesToken_API.json

```diff
@@ -0,0 +1,243 @@
+          {
+            "$ref": "#/parameters/ResourceUriParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Successful response containing the application insights diagnostic services token.",
+            "schema": {
+              "$ref": "#/definitions/DiagnosticServicesTokenResponse"
+            },
+            "examples": {
+              "application/json": {
+                "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get diagnostic services token for resource": {
+            "$ref": "./examples/DiagnosticServicesTokenGetAndPost.json"
+          }
+        }
+      },
+      "post": {
+        "tags": [
+          "DiagnosticServiceToken"
+        ],
+        "operationId": "DiagnosticServiceToken_POST",
+        "description": "**Gets an read-write access token for application insights diagnostic service data.**",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceUriParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/ErrorResponse"
+            }
+          },
+          "200": {
+            "description": "Successful response containing the application insights diagnostic services token.",
+            "schema": {
+              "$ref": "#/definitions/DiagnosticServicesTokenResponse"
+            },
+            "examples": {
+              "application/json": {
+                "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c"
+              }
+            }
+          }
+        },
+        "x-ms-examples": {
+          "Get diagnostic services token for resource": {
+            "$ref": "./examples/DiagnosticServicesTokenGetAndPost.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "DiagnosticServicesTokenResponse": {
+      "type": "object",
+      "properties": {
+        "token": {
```

Comment:
We should tag this with x-ms-secret; so that other first party/third party services do not store this data.

Reply:
@chiragg4u thanks for the suggestion. Once 'x-ms-secret' is added, the token property isn't allowed to be returned in the body. Could you please give me some pointers of fixing that issue?

Reply:
@xiaomi7732 , model validation doesn’t differentiate the HTTP method for secret validation. You could suppress this error in readme.md like this:
```
## Suppression
``` yaml
directive:
  - suppress: SECRET_PROPERTY
    from: xx.json
    where:
      - $.definitions.xx.properties.xx
    reason: Secrets are OK to return in a POST response.
```

Reply:
@raych1 thanks for the feedback. Suppression added.